src/MetadataUtility/Audio/Wave.cs

```diff
@@ -2,12 +2,156 @@
 // All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group.
 // </copyright>
 
-namespace MetadataUtility.Audio.Vendors
+namespace MetadataUtility.Audio
 {
+    using LanguageExt;
+    using LanguageExt.Common;
+    using MetadataUtility.Utilities;
+
     public static class Wave
     {
+        public const string Mime = "audio/wave";
+
+        public const int WaveFileOffset = 7;
+        public const int WaveSampleRateOffset = 24;
+        public const int WaveChannelOffset = 22;
+        public const int WaveBitsPerSampleOffset = 34;
+        public const int WaveFileLengthOffset = 4;
+
         public static readonly byte[] WaveMagicNumber = new byte[] { (byte)'W', (byte)'A', (byte)'V', (byte)'E' };
 
+        public static readonly Error FileTooShort = Error.New("Error reading file: file is not long enough to have a duration header");
+
+        public static readonly Error FileTooShortWave = Error.New("Error reading file: file is not long enough to have a WAVE header");
+
         public static readonly byte[] DataBlockId = new byte[] { (byte)'d', (byte)'a', (byte)'t', (byte)'a' };
+
+        public static Fin<ulong> ReadTotalSamples(FileStream stream)
+        {
+            stream.Seek(WaveFileOffset, SeekOrigin.Begin);
+
+            Span<byte> buffer = stackalloc byte[5];
+            int bytesRead = stream.Read(buffer);
+
+            if (bytesRead != buffer.Length)
+            {
+                return FileTooShort;
+            }
+
+            // flac files have an unsigned 36-bit integer for total sample duration!
+            return BinaryHelpers.Read36BitUnsignedBigEndianIgnoringFirstOctet(buffer);
```

Comment:
So this method is basically incorrect. Have a look at http://soundfile.sapp.org/doc/WaveFormat/ and http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/WAVE.html and https://github.com/QutEcoacoustics/audio-analysis/blob/master/src/Acoustics.Tools/Wav/WavReader.cs .

The `data` chunk is not always guaranteed to be in the same offset. So you have to advance one chunk at a time.

- Read RIFF, advance 4
- Read size, advance 4
- Read WAVE, advance 4
- Read `fmt `, advance 4
- Read `fmt ` length, advance length
- Read next chunk, if chunk is not `data` then advance chunk length
- if chunk is `data` then read number of samples
- then convert that total number of samples to number of inter-channel samples so we report the same value as the FLAC file